internal/pkg/template/jsonnet/function/function.go

```diff
@@ -142,11 +144,18 @@ func ComponentIsAvailable(components *model.ComponentsMap) *jsonnet.NativeFuncti
 }
 
 // SnowflakeWriterComponentID Jsonnet function returns component ID of the Snowflake Writer it the stack.
-func SnowflakeWriterComponentID(components *model.ComponentsMap) *jsonnet.NativeFunction {
+func SnowflakeWriterComponentID(components *model.ComponentsMap, backend ...string) *jsonnet.NativeFunction {
```

Comment:
@michaljurecko I wonder if it is correct to have an "options" parameter for this function or if this parameter should always be required. as for contextual use/load.

1. SnowflakeWriterComponentID(components *model.ComponentsMap, backend ...string)
2. SnowflakeWriterComponentID(components *model.ComponentsMap, backend []string)

We can add it either from templates.requirement or from dependencies.

I don't know exactly how these contexts work.

Reply:
The `backend` argument is now required to make good decision about `componentID`, but I don't understand why you used `backend ...string` instad of `backend []string`.

Reply:
Because this function is used: https://github.com/keboola/keboola-as-code/blob/ab8f731fddb84c4eff9037a0d582d0a1f99becc6/internal/pkg/template/context/load/context.go#L42

Reply:
Ok, so it is logical problem

`load` phase of the template runs on the Templates API startup and when the template update is detected.
- It is not project scoped operation :exclamation: So no backends are available here.
- Or in other words, the `SnowflakeWriterComponentID` was stack scoped, and you want to make it project scoped.
- The templates are pre-loaded so that they can be listed through the API and the template `use` is faster.
- The Snowflake Writer ID is used in `.keboola/repository.json`, in the list of components that a template contains.
  - See https://developers.keboola.com/cli/templates/structure/#snowflake-writer
  
  -----------
  
  
The only solution I can think of is to remove the function from the Load context.

Or refactor the Templates API complexly.

Reply:
But there is a field `requirements` and then the repository.json can include the `backend` or is it something else?

Reply:
What if we just include `[]string{}` backends as parameter? Is it better to just remove it?

Reply:
@Matovidlo yes i can change it and the remove this function from `load context`.

Reply:
@michaljurecko ok, thank for your explanation and your time. 

I will change it and remove it from `load context`

Reply:
Ok, but you should find out where it is used, and agree with the authors of the template.

Reply:
It looks like the placeholder `<keboola.wr-db-snowflake>`, is not used at all 
https://github.com/keboola/keboola-as-code-templates/blob/dde46bc71984e11949e482001a5b685670fa4e77/.keboola/repository.json

So maybe it is safe to remove it at all :man_shrugging: